jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/SHAUtil.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (c) 2019 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at:
+ *
+ *     https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.jkube.kit.common.util;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+public class SHAUtil {
```

Comment:
Most of the utils in this class are already available in the [`AwsSigner4`](https://github.com/eclipse/jkube/blob/03554382b7c292ebffd4133388805fe39cac0ccd/jkube-kit/build/service/docker/src/main/java/org/eclipse/jkube/kit/build/service/docker/auth/ecr/AwsSigner4.java)

We might want to consolidate those prior to continue the work on this PR.

Also, we're reimplementing a read file approach which I'm sure we have elsewehere in the repo too.

Reply:
In general the idea would be to reuse whatever other **core** libraries we rely on to delegate these simple/common tasks

Reply:
You're right. I was able to replace both methods using [Apache commons codec's DigestUtils](https://commons.apache.org/proper/commons-codec/apidocs/org/apache/commons/codec/digest/DigestUtils.html)